scripts/spider/fallback.py

```
@@ -0,0 +1,399 @@
+    if use_proxy:
+        if is_adhoc_mode:
+            state.global_proxy_pool.mark_failure_and_switch()
+        else:
+            if last_failed_html:
+                current_proxy_name = state.global_proxy_pool.get_current_proxy_name()
+                state.save_proxy_ban_html(last_failed_html, current_proxy_name, page_num)
+            for _ in range(PROXY_POOL_MAX_FAILURES):
+                state.global_proxy_pool.mark_failure_and_switch()
+            proxy_was_banned = True
+
+    for attempt in range(max_switches):
```

Ruff (0.15.2)

[warning] Loop control variable `attempt` not used within loop body

Rename unused `attempt` to `_attempt`

(B007)